test/Microsoft.Dnx.Tooling.FunctionalTests/DnuRuntimeRestoreTests.cs

```diff
@@ -0,0 +1,178 @@
+                var lockFilePath = Path.Combine(testDir, "project.lock.json");
+                if (File.Exists(lockFilePath))
+                {
+                    File.Delete(lockFilePath);
+                }
+
+                // Restore the project!
+                var source = Path.Combine(_appEnv.ApplicationBasePath, "..", "..", "misc", "RuntimeRestore", "RuntimeRestoreTestPackage", "feed");
+                DnuTestUtils.ExecDnu(runtimeHomeDir, "restore", $"--source {source}", workingDir: testDir);
+
+                // Check the lock file
+                lockFile = (new LockFileFormat()).Read(Path.Combine(testDir, "project.lock.json"));
+            }
+
+            // We can use the runtime environment to determine the expected RIDs because by default it only uses the current OSes RIDs
+            if (RuntimeEnvironmentHelper.IsWindows)
+            {
+                AssertLockFileTarget(lockFile, "win7-x86", "win7-x86");
+                AssertLockFileTarget(lockFile, "win7-x64", "win7-x64");
+            }
+            else
+            {
+                var osName = RuntimeEnvironmentHelper.RuntimeEnvironment.OperatingSystem.ToLowerInvariant();
+                AssertLockFileTarget(lockFile, osName + "-x86", assemblyRid: null); // There is no linux/darwin-x86 in the test package
+                AssertLockFileTarget(lockFile, osName + "-x64", osName + "-x64");
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(RuntimeComponents))]
+        public void DnuRestore_UsesProjectAndCommandLineProvidedRuntimes(string flavor, string os, string architecture)
+        {
+            var runtimeHomeDir = TestUtils.GetRuntimeHomeDir(flavor, os, architecture);
+
+            LockFile lockFile;
+            using (var testDir = TestUtils.CreateTempDir())
+            {
+                DirTree.CreateFromDirectory(Path.Combine(_appEnv.ApplicationBasePath, "..", "..", "misc", "RuntimeRestore", "TestProject"))
+                    .WriteTo(testDir);
+
+                // Clean up the lock file if it ended up there during the copy
+                var lockFilePath = Path.Combine(testDir, "project.lock.json");
+                if (File.Exists(lockFilePath))
+                {
+                    File.Delete(lockFilePath);
+                }
+
+                // Modify the project
+                AddRuntimeToProject(testDir, "win10-x86");
+
+                // Restore the project!
+                var source = Path.Combine(_appEnv.ApplicationBasePath, "..", "..", "misc", "RuntimeRestore", "RuntimeRestoreTestPackage", "feed");
+                DnuTestUtils.ExecDnu(runtimeHomeDir, "restore", $"--source {source} --runtime linux-x64 --runtime darwin-x64", workingDir: testDir);
+
+                // Check the lock file
+                lockFile = (new LockFileFormat()).Read(Path.Combine(testDir, "project.lock.json"));
+            }
+
+            AssertLockFileTarget(lockFile, "win10-x86", "win8-x86");
```

Comment:
Should we restore for win even though target runtimes were specified in dnu restore explicitly?